### PR TITLE
feat: basic Kiln Prompt Optimizer UI

### PR DESCRIPTION
**Pull request by leonardmq**
## What does this PR do?

Changes:
- add basic GEPA UI
- dependency on server side that does prechecks (before submitting)

## Checklists

- [x] Tests have been run locally and passed
- [x] New tests have been added to any work in /lib

## New Features

* **Kiln Optimizer (GEPA)**: Create and manage prompt optimization jobs with configurable token budgets and evaluation selection.
* **Job Management**: Track optimization job status, view results, and automatically generate optimized prompts upon completion.
* **Evaluation Updates**: Add or modify evaluation names, descriptions, and training set configurations.
* **Navigation**: New Optimizer menu item for quick access to optimization workflows.

### CODE REVIEW

**Comment by leonardmq** [comment]
CI errors carried over from base branch

**Comment by leonardmq** [comment]
@coderabbitai help

**Comment by leonardmq** [comment]
@coderabbitai full review

File: app/desktop/studio_server/gepa_job_api.py [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+                logger.info(
+                    f"Created prompt {prompt.id} from GEPA job {gepa_job.job_id}"
+                )
+
+        gepa_job.save_to_file()
+
+    except Exception as e:
+        logger.error(f"Error updating GEPA job status: {e}", exc_info=True)
+
+    return gepa_job
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply app/desktop/studio_server/gepa_job_api.py:188]: Correct - already addressed in a refactor in https://github.com/Kiln-AI/Kiln/pull/1014 that also derives a run config from the job succeeding

File: app/web_ui/src/routes/(app)/gepa/[project_id]/[task_id]/create_gepa/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,1229 @@
...
+  }
+
+  onMount(async () => {
+    await Promise.all([
+      load_task(),
+      load_task_prompts(project_id, task_id),
+      load_task_run_configs(project_id, task_id),
+      load_evals_and_configs(),
+    ])
+  })
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply app/web_ui/src/routes/(app)/gepa/[project_id]/[task_id]/create_gepa/+page.svelte:589]: Solved in subsequent PR

File: app/desktop/studio_server/eval_api.py [hunk 3]
```diff
@@ -186,6 +186,12 @@ class UpdateFavouriteRequest(BaseModel):
     favourite: bool
 
 
+class UpdateEvalRequest(BaseModel):
+    name: str | None = None
+    description: str | None = None
+    train_set_filter_id: str | None = None
+
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply app/desktop/studio_server/eval_api.py:193]: Is it already getting validated when trying to assign it to the Pydantic model in the endpoint's controller itself though? If so, then P2.

File: app/desktop/studio_server/eval_api.py [hunk 4]
```diff
@@ -365,6 +371,35 @@ async def delete_eval(project_id: str, task_id: str, eval_id: str) -> None:
...
+            # would make comparing results before and after the change very confusing
+            if eval.train_set_filter_id is not None:
+                raise HTTPException(
+                    status_code=400,
+                    detail="Train set filter is already set and cannot be changed. Please create a new eval if you need a different train set.",
+                )
+            eval.train_set_filter_id = request.train_set_filter_id
+
+        eval.save_to_file()
+        return eval
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply app/desktop/studio_server/eval_api.py:401]: Is it already getting validated when trying to assign it to the Pydantic model in the endpoint's controller itself though? If so, then P2.

And actually fine if the filter does not match any entries yet - users could tag task runs with that tag post-hoc. Our tags are not normalized / materialized; they are only derived from whatever set exists

File: app/web_ui/src/routes/(app)/+layout.svelte [hunk 5]
```diff
@@ -268,6 +269,16 @@
...
 
+      <li class="menu-md">
+        <a
+          href={`/gepa/${$ui_state.current_project_id}/${$ui_state.current_task_id}`}
+          class={section == Section.Optimizer ? "active" : ""}
+        >
+          <div class="h-6 w-6 mr-2"></div>
+          Kiln Optimizer</a
+        >
+      </li>
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply app/web_ui/src/routes/(app)/+layout.svelte:280]: Not a problem - subsequent PR reorgs the sidebar
